### PR TITLE
Support custom Vault server configuration

### PR DESCRIPTION
**Pull request by owenfarrell**
As a developer, I want to customize the configuration of my `vault-dev` container so that I can test edge cases and fringe scenarios as necessary.

This change breaks down the existing entrypoint script (`/opt/run.sh`) in to:
* A call to the original entrypoint script provided by the `vault` container. By encapsulating the existing script, this extension does not need to continually adapt to new/changing environmental variables, toggles, etc.
* A configuration script that performs any/all necessary `vault` CLI commands. The default script runs the 2 commands included in `run.sh` to set up secret engines accordingly. Separating this out in to its own script allows consumers to extend or override the default commands.
* An initialization script that orchestrates the sequence of the Vault configuration script and initial data loads. 

Additionally, there are a couple of polishing commits included in this PR (tab stops -> spaces, renaming files to simplify the Dockerfile).

### CODE REVIEW

**Comment by bkeroackdsc** [comment]
Hi @owenfarrell ! Thank you for the contribution.

Offhand it looks reasonable, but can you please revert your changes to the `.helm` folder? This is where we store in-repository Helm charts (by internal convention).

**Comment by owenfarrell** [comment]
@bkeroackdsc - Thanks for the review!

Good call on the Helm updates. Since I'm not trying to use this container in a Kubernetes context, I just blindly moved all of the files that were landing in `/tmp` to the `tmp` subfolder. I also cleaned up the file permissions for the scripts in the index.

Let me know if you have any other feedback!

**Comment by owenfarrell** [comment]
@bkeroackdsc - Any other feedback on this PR?

**Comment by bkeroackdsc** [comment]
@owenfarrell Thanks for the reminder. It looks like you still have some files moved into `/tmp` in this PR. Please revert that and we will do a final CR.

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 
-ADD . /tmp/
+ARG ENTRYPOINT_DIR=/usr/local/bin
 
-RUN /tmp/setup.sh
+RUN apk update --no-cache \
+  && apk add --no-cache jq \
+  && mv ${ENTRYPOINT_DIR}/docker-entrypoint.sh ${ENTRYPOINT_DIR}/original-entrypoint.sh
+
+ADD tmp /tmp/
```
Comment (owenfarrell) [inline Dockerfile:9]: @bkeroackdsc - Moving the files in to a `tmp` folder was an intentional change since I'm also adding new files. My thought here was to avoid the two-step copy (add to `/tmp`, move to destination) while maintaining backwards compatibility.

But I recognize that your team has more context on how this image is being used internally. So happy to adjust accordingly.
